### PR TITLE
[diffutils] Bump to v3.8

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: D/diffutils/build_tarballs.jl [hunk 1]
```diff
@@ -3,24 +3,27 @@
...
 script = raw"""
 cd $WORKSPACE/srcdir/diffutils-*/
+
 if [[ "${target}" == *-mingw* ]]; then
-    atomic_patch -p1 ../patches/win_signal_handling.patch
+    atomic_patch -p1 ../patches/win_sa_restart.patch
 fi
-./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --target=${compiler_target} --disable-dependency-tracking
```
Comment (giordano) [inline D/diffutils/build_tarballs.jl:25]: Where did you even find `compiler_target`?  I doesn't exists.

Comment (jeremiahpslewis) [reply D/diffutils/build_tarballs.jl:26]: hahaha

Comment (jeremiahpslewis) [reply D/diffutils/build_tarballs.jl]: thanks.

File: D/diffutils/build_tarballs.jl [hunk 2]
```diff
@@ -3,24 +3,28 @@
 using BinaryBuilder, Pkg
 
 name = "diffutils"
-version = v"3.7.0"
+version = v"3.8"
```
Comment (giordano) [inline D/diffutils/build_tarballs.jl:6]:
```suggestion
version_string = "3.8"
version = VersionNumber(version_string)
```

File: D/diffutils/build_tarballs.jl [hunk 3]
```diff
@@ -3,24 +3,28 @@
...
 
 # Collection of sources required to complete build
 sources = [
     ArchiveSource(
-        "https://ftp.gnu.org/gnu/diffutils/diffutils-3.7.tar.xz",
-        "b3a7a6221c3dc916085f0d205abf6b8e1ba443d4dd965118da364a1dc1cb3a26",
+        "https://ftp.gnu.org/gnu/diffutils/diffutils-$(version.major).$(version.minor).tar.xz",
```
Comment (giordano) [inline D/diffutils/build_tarballs.jl:11]:
```suggestion
        "https://ftp.gnu.org/gnu/diffutils/diffutils-$(version_string).tar.xz",
```

**Pull request merged**